### ISSUE TITLE
Using updated `fetch-depth` and `fetch-tags` settings fail for `actions/checkout` when processing release builds

### ISSUE THREAD

**Issue by atc0005** [atc0005/shared-project-resources#156]
## Overview

After much hair pulling I seemed to get all workflows *working* with a combination of settings that allowed pulling a subset of history associated with the current PR branch *with tags* and then a subset of history from both primary and development branches. This allowed the now standard "PR branch is updated with source branch" assertions to complete successfully.

Then I ran into issues with the automated release workflows failing:

```log
Run actions/checkout@v3.6.0
  with:
    fetch-depth: 50
    fetch-tags: true
    repository: atc0005/check-vmware
    token: ***
    ssh-strict: true
    persist-credentials: true
    clean: true
    sparse-checkout-cone-mode: true
    lfs: false
    submodules: false
    set-safe-directory: true
/usr/bin/docker exec  0c1e1aa8b7187e30eb0f57e7ef0daa94a0766bf10a8761f906bad309b2b831cb sh -c "cat /etc/*release | grep ^ID"
Syncing repository: atc0005/check-vmware
Getting Git version info
Temporarily overriding HOME='/__w/_temp/baf1c9be-a3df-4e8c-9901-e47d9ed059d3' before making global git config changes
Adding repository directory to the temporary git global config as a safe directory
/usr/bin/git config --global --add safe.directory /__w/check-vmware/check-vmware
Deleting the contents of '/__w/check-vmware/check-vmware'
Initializing the repository
  /usr/bin/git init /__w/check-vmware/check-vmware
  hint: Using 'master' as the name for the initial branch. This default branch name
  hint: is subject to change. To configure the initial branch name to use in all
  hint: of your new repositories, which will suppress this warning, call:
  hint: 
  hint: 	git config --global init.defaultBranch <name>
  hint: 
  hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
  hint: 'development'. The just-created branch can be renamed via this command:
  hint: 
  hint: 	git branch -m <name>
  Initialized empty Git repository in /__w/check-vmware/check-vmware/.git/
  /usr/bin/git remote add origin https://github.com/atc0005/check-vmware
Disabling automatic garbage collection
Setting up auth
  /usr/bin/git config --local --name-only --get-regexp core\.sshCommand
  /usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
  /usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
  /usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
  /usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic ***
Fetching the repository
  /usr/bin/git -c protocol.version=2 fetch --prune --progress --no-recurse-submodules --depth=50 origin +3ab151278eb9c5925b43e42b3cbbf4e85141c18c:refs/tags/v0.36.0-alpha.3
  Error: fatal: Cannot fetch both 3ab151278eb9c5925b43e42b3cbbf4e85141c18c and refs/tags/v0.36.0-alpha.3 to refs/tags/v0.36.0-alpha.3
  The process '/usr/bin/git' failed with exit code 128
  Waiting 19 seconds before trying again
  /usr/bin/git -c protocol.version=2 fetch --prune --progress --no-recurse-submodules --depth=50 origin +3ab151278eb9c5925b43e42b3cbbf4e85141c18c:refs/tags/v0.36.0-alpha.3
  Error: fatal: Cannot fetch both 3ab151278eb9c5925b43e42b3cbbf4e85141c18c and refs/tags/v0.36.0-alpha.3 to refs/tags/v0.36.0-alpha.3
  The process '/usr/bin/git' failed with exit code 128
  Waiting 15 seconds before trying again
  /usr/bin/git -c protocol.version=2 fetch --prune --progress --no-recurse-submodules --depth=50 origin +3ab151278eb9c5925b43e42b3cbbf4e85141c18c:refs/tags/v0.36.0-alpha.3
  Error: fatal: Cannot fetch both 3ab151278eb9c5925b43e42b3cbbf4e85141c18c and refs/tags/v0.36.0-alpha.3 to refs/tags/v0.36.0-alpha.3
  Error: The process '/usr/bin/git' failed with exit code 128
```

This line seems to be particularly problematic:

> ` /usr/bin/git -c protocol.version=2 fetch --prune --progress --no-recurse-submodules --depth=50 origin +3ab151278eb9c5925b43e42b3cbbf4e85141c18c:refs/tags/v0.36.0-alpha.3`

Commit `3ab151278eb9c5925b43e42b3cbbf4e85141c18c` is for the annotated tag `v0.36.0-alpha.3`. I don't know if a non-annotated tag would produce the same issue.

Yesterday before the changes for GH-148 this was the line used:

> `/usr/bin/git -c protocol.version=2 fetch --prune --progress --no-recurse-submodules origin +refs/heads/*:refs/remotes/origin/* +refs/tags/*:refs/tags/*`

I'm going to assume that the difference is that previously we fetched *all* history.

At this point I'm strongly considering reverting all changes made towards GH-148.

## References

- https://github.com/atc0005/shared-project-resources/issues/148
- https://github.com/atc0005/check-vmware/actions/runs/6023872015/job/16341488361#logs
  - successful job (day before)
- https://github.com/atc0005/check-vmware/actions/runs/6036206525/job/16378451759#logs
  - failing job (today)
  - before cedd8fcc8109222506502925b228a45f00284a5d (partial fix) and 68911c803fd642cd99fd9f8004d5e973850bb684 (remaining fix)
- https://github.com/atc0005/check-vmware/actions/runs/6036206525/job/16380384208
  - successful job (today)
  - after 68911c803fd642cd99fd9f8004d5e973850bb684 (remaining fix)

**Comment by atc0005** [comment]
Resolved via cedd8fcc8109222506502925b228a45f00284a5d (partial fix) and 68911c803fd642cd99fd9f8004d5e973850bb684 (remaining fix).

**Comment by atc0005** [comment]
Not resolved.

https://github.com/atc0005/check-vmware/actions/runs/6036206525/job/16380384208

Found this in the deb/rpm generation output:

```log
created package: /builds/release_assets/packages/dev/check-vmware-dev-v0.0.0-dev.0.g3ab1512_amd64.deb

  - dev RPM package ...
using rpm packager...
created package: /builds/release_assets/packages/dev/check-vmware-dev-v0.0.0-dev.0.g3ab1512.x86_64.rpm
```

The release version generation logic isn't working. I will need to amend the checkout logic to provide needed context, evidently.

**Comment by atc0005** [comment]
Resolved (finally) by the work on #159 (commit 45f2d5fbbc6bd326ddd97e66c32539c3f08bb887).

**Comment by atc0005** [comment]
> Resolved (finally) by the work on #159 (commit [45f2d5f](https://github.com/atc0005/shared-project-resources/commit/45f2d5fbbc6bd326ddd97e66c32539c3f08bb887)).

This specific problem is resolved, but other issues with recent changes remain. Will file separate GH issues for those.
